src/main/java/build/buildfarm/common/distributed/DistributedMap.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.common.distributed;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @class DistributedMap
+ * @brief A distributed map.
+ * @details A distributed map is an implementation of a map data structure which can be shared
+ *     across a distributed system. For example, implementations of this interface might choose to
+ *     use redis, hazelcast, postgres, etc.
+ */
+public interface DistributedMap<DistributedClient> {
```

Comment:
This is a Map with an extra parameter added to every method. It only makes sense if you're trying to refer to a different client in the same instance Map, which I rather doubt, since you're hiding implementations for it in the concretes. The only different method from Map is the timeout insert, which still doesn't have a checked path, and the details for which can be hidden in a concrete class implementation, unless again, you want to have a different timeout for different circumstances.

In my blatantly biased opinion, you don't need this.

Reply:
Are you suggesting we hold a copy of `JedisCluster` inside the implementation of `RedisHashMap` instead?  That seems like the only way I can inherit from `Map<String, String>` and remove it as an input parameter from all the methods.

Reply:
Precisely. Make this an inherited implementation of Map<String, String>